### ISSUE TITLE
crashes Julia

### ISSUE THREAD

**Issue by nsajko** [JuliaLang/AllocCheck.jl#67]
```
$ ./julia -g2
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.11.0-DEV.1274 (2024-01-14)
 _/ |\__'_|_|_|\__'_|  |  Commit 681816cedf6 (0 days old master)
|__/                   |

julia> using AllocCheck

(@v1.11) pkg> st AllocCheck
Status `~/.julia/environments/v1.11/Project.toml`
  [9b6a8646] AllocCheck v0.1.0 `https://github.com/JuliaLang/AllocCheck.jl#main`

julia> versioninfo()
Julia Version 1.11.0-DEV.1274
Commit 681816cedf6 (2024-01-14 15:03 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 8 × AMD Ryzen 3 5300U with Radeon Graphics
  WORD_SIZE: 64
  LLVM: libLLVM-16.0.6 (ORCJIT, znver2)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)
Environment:
  JULIA_NUM_PRECOMPILE_TASKS = 3
  JULIA_PKG_PRECOMPILE_AUTO = 0

julia> check_allocs(sin, Tuple{Float64})

[73333] signal 11 (1): Segmentation fault
in expression starting at REPL[3]:1
_ZNK4llvm5Value10getContextEv at /home/nsajko/tmp/jl/jl/julia-681816cedf/bin/../lib/julia/libLLVM-16jl.so (unknown line)
LLVMGetValueContext at /home/nsajko/.julia/packages/LLVM/OLMpi/lib/libLLVM_extra.jl:100 [inlined]
context at /home/nsajko/.julia/packages/LLVM/OLMpi/src/core/value.jl:108 [inlined]
BasicBlock at /home/nsajko/.julia/packages/LLVM/OLMpi/src/core/basicblock.jl:22 [inlined]
rename_calls_and_throws! at /home/nsajko/.julia/packages/AllocCheck/bvNS7/src/AllocCheck.jl:29
#find_allocs!#37 at /home/nsajko/.julia/packages/AllocCheck/bvNS7/src/AllocCheck.jl:105
find_allocs! at /home/nsajko/.julia/packages/AllocCheck/bvNS7/src/AllocCheck.jl:96
unknown function (ip: 0x7fa663114d8d)
#39 at /home/nsajko/.julia/packages/AllocCheck/bvNS7/src/AllocCheck.jl:210
JuliaContext at /home/nsajko/.julia/packages/GPUCompiler/2mJjc/src/driver.jl:47
#check_allocs#38 at /home/nsajko/.julia/packages/AllocCheck/bvNS7/src/AllocCheck.jl:206
check_allocs at /home/nsajko/.julia/packages/AllocCheck/bvNS7/src/AllocCheck.jl:199
unknown function (ip: 0x7fa691d309a6)
jl_apply at /cache/build/tester-amdci5-8/julialang/julia-master/src/julia.h:2153 [inlined]
do_call at /cache/build/tester-amdci5-8/julialang/julia-master/src/interpreter.c:126
eval_value at /cache/build/tester-amdci5-8/julialang/julia-master/src/interpreter.c:223
eval_stmt_value at /cache/build/tester-amdci5-8/julialang/julia-master/src/interpreter.c:174 [inlined]
eval_body at /cache/build/tester-amdci5-8/julialang/julia-master/src/interpreter.c:675
jl_interpret_toplevel_thunk at /cache/build/tester-amdci5-8/julialang/julia-master/src/interpreter.c:815
jl_toplevel_eval_flex at /cache/build/tester-amdci5-8/julialang/julia-master/src/toplevel.c:941
jl_toplevel_eval_flex at /cache/build/tester-amdci5-8/julialang/julia-master/src/toplevel.c:884
ijl_toplevel_eval_in at /cache/build/tester-amdci5-8/julialang/julia-master/src/toplevel.c:992
eval at ./boot.jl:428 [inlined]
eval_user_input at /home/nsajko/tmp/jl/jl/julia-681816cedf/share/julia/stdlib/v1.11/REPL/src/REPL.jl:154
repl_backend_loop at /home/nsajko/tmp/jl/jl/julia-681816cedf/share/julia/stdlib/v1.11/REPL/src/REPL.jl:250
#start_repl_backend#48 at /home/nsajko/tmp/jl/jl/julia-681816cedf/share/julia/stdlib/v1.11/REPL/src/REPL.jl:235
start_repl_backend at /home/nsajko/tmp/jl/jl/julia-681816cedf/share/julia/stdlib/v1.11/REPL/src/REPL.jl:232
#run_repl#61 at /home/nsajko/tmp/jl/jl/julia-681816cedf/share/julia/stdlib/v1.11/REPL/src/REPL.jl:391
run_repl at /home/nsajko/tmp/jl/jl/julia-681816cedf/share/julia/stdlib/v1.11/REPL/src/REPL.jl:377
jfptr_run_repl_12750 at /home/nsajko/.julia/compiled/v1.11/REPL/u0gqU_JNfbi.so (unknown line)
#1096 at ./client.jl:440
jfptr_YY.1096_16855 at /home/nsajko/.julia/compiled/v1.11/REPL/u0gqU_JNfbi.so (unknown line)
jl_apply at /cache/build/tester-amdci5-8/julialang/julia-master/src/julia.h:2153 [inlined]
jl_f__call_latest at /cache/build/tester-amdci5-8/julialang/julia-master/src/builtins.c:875
#invokelatest#2 at ./essentials.jl:955 [inlined]
invokelatest at ./essentials.jl:952 [inlined]
run_main_repl at ./client.jl:424
repl_main at ./client.jl:561 [inlined]
_start at ./client.jl:535
jfptr__start_65272.1 at /home/nsajko/tmp/jl/jl/julia-681816cedf/lib/julia/sys.so (unknown line)
jl_apply at /cache/build/tester-amdci5-8/julialang/julia-master/src/julia.h:2153 [inlined]
true_main at /cache/build/tester-amdci5-8/julialang/julia-master/src/jlapi.c:947
jl_repl_entrypoint at /cache/build/tester-amdci5-8/julialang/julia-master/src/jlapi.c:1106
main at /cache/build/tester-amdci5-8/julialang/julia-master/cli/loader_exe.c:58
unknown function (ip: 0x7fa698ffbccf)
__libc_start_main at /usr/lib/libc.so.6 (unknown line)
unknown function (ip: 0x4010b8)
Allocations: 28118316 (Pool: 28116278; Big: 2038); GC: 19
Segmentation fault (core dumped)
```

Maybe related to the move to LLVM 16? Just guessing.

**Comment by topolarity** [comment]
Definitely related to the LLVM 16 bump - We'll have to wait a bit for the dependency chain to be ready for us here, but this only affects nightly for now

**Comment by nsajko** [comment]
FTR it seems that Julia 1.11 may actually end up shipping with LLVM 17: https://github.com/JuliaLang/julia/pull/53070

**Comment by topolarity** [comment]
I believe this should be resolved in v0.2.0 - Please re-open if not
